addons/account/models/account_move_send.py

```diff
@@ -183,7 +183,7 @@ def _get_default_mail_partner_ids(self, move, mail_template, mail_lang):
             partner_to = self._get_mail_default_field_value_from_template(mail_template, mail_lang, move, 'partner_to')
             partner_ids = mail_template._parse_partner_to(partner_to)
             partners |= self.env['res.partner'].sudo().browse(partner_ids).exists()
-        return partners.filtered('email')
+        return partners if self._context.get('partners_without_mail') else partners.filtered('email')
```

Comment:
If we don't have a choice to use the context to do that. I'd prefer you use the usual way of extracting it: `self.env.context`.